remove WATER_MAP_TEST job spec (#1904)
ASFHyP3/hyp3: CHANGELOG.md

and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [4.3.0]
### Changed

> - The `iterative_min_size` and `minimization_metric` parameters have been added to the `WATER_MAP` job spec, deprecating the `WATER_MAP_TEST` job spec. Notably, the default `minimization_metric` is now `ts`, instead of `fmi` as set in `WATER_MAP_TEST`

Reviewer: Actually, I just realized the `WATER_MAP_TEST` job spec has been removed, not deprecated. So maybe something more like this?
```suggestion
- The `iterative_min_size` and `minimization_metric` parameters have been moved from the `WATER_MAP_TEST` job spec to the `WATER_MAP` job spec. The default `minimization_metric` value has been changed from `fmi` to `ts`.
```

Author: @jtherrmann oh I like that. I couldn't figure out how to word the removal and updates.

> - The `iterative_min_size` and `minimization_metric` parameters have been added to the `WATER_MAP` job spec, depreciating the `WATER_MAP_TEST` job spec. Notably, the default `minimization_metric` is now `ts`, instead of `fmi` as set in `WATER_MAP_TEST`

Reviewer: ```suggestion
- The `iterative_min_size` and `minimization_metric` parameters have been added to the `WATER_MAP` job spec, deprecating the `WATER_MAP_TEST` job spec. Notably, the default `minimization_metric` is now `ts`, instead of `fmi` as set in `WATER_MAP_TEST`
```

Author: I just *cannot* keep deprecate and depreciate straight :rofl:
